sqlglot/expressions.py

```diff
@@ -3538,14 +3540,16 @@ class Case(Func):
     arg_types = {"this": False, "ifs": True, "default": False}
 
     def when(self, condition: ExpOrStr, then: ExpOrStr, copy: bool = True, **opts) -> Case:
-        this = self.copy() if copy else self
-        this.append("ifs", If(this=maybe_parse(condition, **opts), true=maybe_parse(then, **opts)))
-        return this
+        instance = _maybe_copy(self, copy)
+        instance.append(
+            "ifs", If(this=maybe_parse(condition, **opts), true=maybe_parse(then, **opts))
```

Comment:
Here we don't copy `condition` and `then` because `copy` is not included in `opts`. Should we default to copying these expressions, i.e. pass in `**{**opts, copy: True}` to `maybe_parse`?

Reply:
why don't you just pass copy in?

Reply:
Done.